src/common.rs

```diff
@@ -222,8 +222,13 @@ impl Frame<'static> {
         for pix in pixels.chunks_exact_mut(4) {
             if pix[3] != 0 {
                 pix[3] = 0xFF;
-            } else {
+            } else if transparent.is_none() {
                 transparent = Some([pix[0], pix[1], pix[2], pix[3]]);
+            } else if let Some([r, g, b, a]) = transparent {
+                pix[0] = r;
+                pix[1] = g;
+                pix[2] = b;
+                pix[3] = a;
```

Comment:
I'd use `if let` instead of match here since we only need contents from the `Some`-branch but follow https://github.com/image-rs/image-gif/pull/224#issuecomment-3564519869 otherwise.

```suggestion
                continue;
            }
            
            if let Some([r, g, b, a]) = transparent {
                pix[0] = r;
                pix[1] = g;
                pix[2] = b;
                pix[3] = a;
            } else {
                transparent = Some([pix[0], pix[1], pix[2], pix[3]]);
            }
```